src/JsonProducer.php

```diff
@@ -48,6 +50,16 @@ public function publish(
     ) {
         $attributes = array_merge_recursive($this->defaultAttributes, $attributes);
 
-        $this->exchange->publish(json_encode($message), $routingKey, $flags, $attributes);
+        try {
+            $message = json_encode($message);
+        } catch (\Throwable $e) {
```

Comment:
`json_encode` does not throw an exception, but an object implementing `JsonSerializable` can throw an exception in its `jsonSerialize` method.